src/mongocxx/test/client_side_encryption.cpp

```diff
@@ -1504,27 +1524,117 @@ TEST_CASE("Custom endpoint", "[client_side_encryption]") {
     }
 }
 
+void bypass_mongocrypt_via_shared_library(const std::string& shared_lib_path,
+                                          bsoncxx::document::view_or_value external_schema) {
+    // Via loading shared library
+    // The following tests that loading crypt_shared bypasses spawning mongocryptd.
+    //
+    // Note
+    //
+    // IMPORTANT: This test requires the crypt_shared library be loaded. If the crypt_shared
+    // library is not available, skip the test.
+    //
+
+    // 1. Create a MongoClient configured with auto encryption (referred to as client_encrypted)
+    //
+    // Configure the required options. Use the local KMS provider as follows:
+    //
+    // { "local": { "key": <base64 decoding of LOCAL_MASTERKEY> } }
+    // Configure with the keyVaultNamespace set to keyvault.datakeys.
+    //
+    // Configure client_encrypted to use the schema external/external-schema.json for db.coll by
+    // setting a schema map like: { "db.coll": <contents of external-schema.json>}
+    //
+    // Configure the following extraOptions:
+    //
+    // {
+    //   "mongocryptdURI": "mongodb://localhost:27021/db?serverSelectionTimeoutMS=1000",
+    //   "mongocryptdSpawnArgs": [
+    //      "--pidfilepath=bypass-spawning-mongocryptd.pid",
+    //      "--port=27021"
+    //   ],
+    //   "cryptSharedLibPath": "<path to shared library>",
+    //   "cryptSharedLibRequired": true
+    // }
+    auto extra = make_document(
+        kvp("mongocryptdURI", "mongodb://localhost:27021/db?serverSelectionTimeoutMS=1000"),
+        kvp("mongocryptdSpawnArgs",
+            make_array("--pidfilepath=bypass-spawning-mongocryptd.pid", "--port=27021")),
+        kvp("cryptSharedLibPath", shared_lib_path),
+        kvp("cryptSharedLibRequired", true));
+
+    options::auto_encryption auto_encrypt_opts{};
+    auto_encrypt_opts.kms_providers(_make_kms_doc());
+    auto_encrypt_opts.tls_opts(_make_tls_opts());
+    auto_encrypt_opts.key_vault_namespace({"keyvault", "datakeys"});
+    auto_encrypt_opts.schema_map({external_schema.view()});
+    auto_encrypt_opts.extra_options({extra.view()});
+
+    options::client client_encrypted_opts;
+    client_encrypted_opts.auto_encryption_opts(std::move(auto_encrypt_opts));
+
+    class client client_encrypted {
+        uri{}, test_util::add_test_server_api(client_encrypted_opts),
+    };
+
+    // 2. Use client_encrypted to insert the document {"unencrypted": "test"} into db.coll.
+    // Expect this to succeed.
+    auto coll = client_encrypted["db"]["coll"];
+    coll.insert_one(make_document(kvp("encrypted", "test")));
```

Comment:
```suggestion
    coll.insert_one(make_document(kvp("unencrypted", "test")));
```

Use `unencrypted` to match the test.

Aside: The motivation for using the field `unencrypted` instead of the `encrypted` is to simplify test setup. Inserting `unencrypted` still sends the command to Query Analysis (mongocryptd / crypt_shared). But no encryption is required. So the test does not need to insert the necessary key document to encrypt.